packages/connect/src/data/firmwareInfo.ts

```diff
@@ -44,220 +71,228 @@ export const getOnlineReleases = async (internalModel: DeviceModelInternal) => {
     return [] as FirmwareRelease[];
 };
 
-const getChangelog = (releases2: FirmwareRelease[], features: StrictFeatures) => {
-    // releases are already filtered, so they can be considered "safe".
-    // so lets build changelog! It should include only those firmwares, that are
-    // newer than currently installed firmware.
+export const parseFirmwareReleaseConfig = (config: FirmwareReleaseConfig) => {
+    firmwareReleasesConfig = config.releases;
+    firmwareIntermediaryReleasesConfig = config.intermediaries;
 
-    if (features.bootloader_mode) {
-        // the problem with bootloader is that we see only bootloader and not firmware version
-        // and multiple releases may share same bootloader version. we really can not tell that
-        // the versions that are installable are newer. so...
-        if (features.firmware_present && features.major_version === 1) {
-            // return null signaling that we don't really know, but only if some firmware
-            // is already installed!
-            return null;
-        }
-        if (features.firmware_present && features.major_version === 2) {
-            // little different situation is with model 2, where in bootloader (and with some fw installed)
-            // we actually know the firmware version
-            return releases2.filter(r =>
-                versionUtils.isNewer(r.version, [
-                    features.fw_major,
-                    features.fw_minor,
-                    features.fw_patch,
-                ]),
-            );
-        }
-
-        // for fresh devices, we can assume that all releases are actually "new"
-        return releases2;
-    }
-
-    // otherwise we are in firmware mode and because each release in releases list has
-    // version higher than the previous one, we can filter out the version that is already
-    // installed and show only what's new!
-    return releases2.filter(r =>
-        versionUtils.isNewer(r.version, [
-            features.major_version,
-            features.minor_version,
-            features.patch_version,
-        ]),
-    );
+    return {
+        releases,
+        intermediaryReleases: firmwareIntermediaryReleasesConfig,
+    };
 };
 
-const isNewer = (release: FirmwareRelease, features: StrictFeatures) => {
-    if (features.major_version === 1 && features.bootloader_mode) {
-        return null;
-    }
+const getCurrentFirmwareVersion = (features: Features): VersionArray => {
```

Comment:
I think we already had this discussion but some older model Ones can't tell firmware version in the bootloader. 

Also maybe some comments to this fn would be nice. Code looks cryptic for people not familiar about this topic.

Reply:
Sure, I am adding some comment https://github.com/trezor/trezor-suite/pull/18821/commits/e4dbafe33c4a43d37f5792e28afb7d999b480391#diff-c30ddd912ad02dc9dace968df9d44aebc836ee02b76e3da843475b5478116d18R90

But also thinking about the function again, the way it was done was very confusing for me so I was trying to make it better for understanding. Now I am looking into it in a different way.